airflow-core/docs/howto/deadline-alerts.rst

```diff
@@ -0,0 +1,263 @@
+
+::
+
+    |------|----------|---------|------------|--------|
+         Queued     Start    Deadline    Reference
+         09:15      09:17     09:30       10:00
+
+.. note::
+    Note that since the interval is a negative value, the deadline is before the reference in this case.
+
+Using Callbacks
+---------------
+
+When a deadline is exceeded, the callback is executed. You can use any async :doc:`Notifier </howto/notifications>`
+or create a custom callback function.
+
+Using Built-in Notifiers
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here's an example using the Slack notifier if the DagRun has not finished within 30 minutes of it being queued:
+
+.. code-block:: python
+
+    with DAG(
+        dag_id="slack_deadline_alert",
+        deadline=DeadlineAlert(
+            reference=DeadlineReference.DAGRUN_QUEUED_AT,
+            interval=timedelta(minutes=30),
+            callback=SlackNotifier(
+                slack_conn_id="slack_default",
+                channel="#alerts",
+                text="DAG {{ dag.dag_id }} has been running for more than 30 minutes since being queued.",
+                username="Airflow Alerts",
+            ),
+        ),
+    ):
+        EmptyOperator(task_id="example_task")
+
+Creating Custom Callbacks
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can create custom callbacks for more complex handling. The ``callback_kwargs`` specified in
+the ``DeadlineAlert`` are passed to the callback function.
+
+.. code-block:: python
+
+    # Place this method in `/files/plugins/deadline_callbacks.py`
+    async def custom_callback(**kwargs):
+        """Handle deadline violation with custom logic."""
+        print(f"Deadline exceeded for DAG {kwargs.get("dag_id")}!")
+        print(f"Alert type: {kwargs.get("alert_type")}")
+        # Additional custom handling here
+
+
+    # Place this in a dag file
+    from datetime import timedelta
+
+    from deadline_callbacks import custom_callback
+
+    from airflow import DAG
+    from airflow.providers.standard.operators.empty import EmptyOperator
+    from airflow.sdk.definitions.deadline import DeadlineAlert, DeadlineReference
+
+    with DAG(
+        dag_id="custom_deadline_alert",
+        deadline=DeadlineAlert(
+            reference=DeadlineReference.DAGRUN_QUEUED_AT,
+            interval=timedelta(minutes=15),
+            callback=custom_callback,
+            callback_kwargs={"alert_type": "time_exceeded", "dag_id": "custom_deadline_alert"},
+        ),
+    ):
+        EmptyOperator(task_id="example_task")
+
+.. note::
+    Regarding Deadline callbacks:
+
+    * Async callbacks are recommended as they will be executed by the Triggerer.
+    * Users must ensure any async callback is importable by the Triggerer.
+    * One easy way to do this is to place the callback as a top-level method in a new file in the plugins folder.
+    * The Triggerer may need to be restarted when a callback is added or changed in order to reload the files.
+
+
+Deadline Calculation
```

Comment:
This block is really nice to hit home the notion of how this all works. Maybe even pull it closer to the top.